src/Sulu/Component/Rest/ListBuilder/Metadata/ListXmlLoader.php

```diff
@@ -225,17 +231,29 @@ private function loadConcatenationPropertyMetadata(\DOMXPath $xpath, \DOMNode $p
     /**
      * Extracts filter type parameters from dom-node.
      *
-     * @return array
+     * @return ?array
      */
-    protected function getFilterTypeParameters(\DOMXPath $xpath, \DOMNode $propertyNode)
+    protected function getFilterTypeParameters(\DOMXPath $xpath, \DOMNode $filterNode)
     {
         $parameters = [];
-        foreach ($xpath->query('x:filter-type-parameters/x:parameter', $propertyNode) as $parameterNode) {
-            $key = XmlUtil::getValueFromXPath('@key', $xpath, $parameterNode);
-            $parameters[$key] = $this->parameterBag->resolveValue(trim($parameterNode->nodeValue));
+        foreach ($xpath->query('x:param', $filterNode) as $paramNode) {
+            $name = XmlUtil::getValueFromXPath('@name', $xpath, $paramNode);
+            $type = XmlUtil::getValueFromXPath('@type', $xpath, $paramNode);
+
+            if ('collection' === $type) {
+                $parameters[$name] = $this->getFilterTypeParameters($xpath, $paramNode);
+            } else {
+                $parameters[$name] = $this->parameterBag->resolveValue(
```

Comment:
i this will behave kind of strange if the `name` attribute of a param is not set